### PR TITLE
DTD termdet fix

### PR DESCRIPTION
**Pull request by therault**
I could run the DTD tests on my laptop in heavy debug mode with these changes.

Opening it up for review and testing for the release.

### CODE REVIEW

File: parsec/data_dist/matrix/redistribute/redistribute_dtd.c [hunk 1]
```diff
@@ -416,11 +416,11 @@ parsec_redistribute_New_dtd(parsec_context_t *parsec,
...
 
     /* Finishing all the tasks inserted, but not finishing the handle */
-    int rc = parsec_dtd_taskpool_wait( dtd_tp );
-    PARSEC_CHECK_ERROR(rc, "parsec_dtd_taskpool_wait");
+    int rc = parsec_taskpool_wait( dtd_tp );
+    PARSEC_CHECK_ERROR(rc, "parsec_taskpool_wait");
 
     /* Cleaning up the parsec handle */
-    parsec_taskpool_free( dtd_tp );
+    parsec_dtd_taskpool_release( dtd_tp );
```
Comment (abouteiller) [inline parsec/data_dist/matrix/redistribute/redistribute_dtd.c:423]: I thought we wanted to converge all these things to a single API

**Comment by abouteiller** [comment]
I get the following ctest crash, all dplasma dtd testers (see https://github.com/ICLDisco/dplasma/pull/64)  deadlock

```
103: [c07:23149:0:23149] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x70)
103: Rank: 6, core 2: entering PTG task (22, 9) of TP ptg_tp (2/0x934880), sleeping for 1 seconds
103: Rank: 6, core 4: leaving PTG task (22, 16) of TP ptg_tp (2/0x934880), slept 1 seconds
103: Rank: 6, core 4: entering PTG task (22, 8) of TP ptg_tp (2/0x934880), sleeping for 1 seconds
103: Rank: 6, core 3: leaving PTG task (22, 15) of TP ptg_tp (2/0x934880), slept 1 seconds
103: Rank: 6, core 3: entering PTG task (22, 7) of TP ptg_tp (2/0x934880), sleeping for 1 seconds
103: Rank: 6, core 5: leaving PTG task (22, 14) of TP ptg_tp (2/0x934880), slept 1 seconds
103: Rank: 6, core 5: entering PTG task (22, 6) of TP ptg_tp (2/0x934880), sleeping for 1 seconds
103: Rank: 6, core 0: leaving PTG task (22, 13) of TP ptg_tp (2/0x934880), slept 1 seconds
103: Rank: 6, core 0: entering PTG task (22, 5) of TP ptg_tp (2/0x934880), sleeping for 1 seconds
103: Rank: 6, core 1: leaving PTG task (22, 12) of TP ptg_tp (2/0x934880), slept 1 seconds
103: Rank: 6, core 7: leaving PTG task (22, 11) of TP ptg_tp (2/0x934880), slept 1 seconds
103: ==== backtrace (tid:  24966) ====
103:  0  /sw/spack/2022-02-10/opt/spack/linux-scientific7-x86_64/gcc-7.3.0/ucx-1.11.2-eu33bafslc3brefgjapdl6ilwpnhhyh4/lib/libucs.so.0(ucs_handle_error+0xe4) [0x7ffff4775e14]
103:  1  /sw/spack/2022-02-10/opt/spack/linux-scientific7-x86_64/gcc-7.3.0/ucx-1.11.2-eu33bafslc3brefgjapdl6ilwpnhhyh4/lib/libucs.so.0(+0x2b13c) [0x7ffff477613c]
103:  2  /sw/spack/2022-02-10/opt/spack/linux-scientific7-x86_64/gcc-7.3.0/ucx-1.11.2-eu33bafslc3brefgjapdl6ilwpnhhyh4/lib/libucs.so.0(+0x2b3a4) [0x7ffff47763a4]
103:  3  /usr/lib64/libpthread.so.0(+0xf630) [0x7ffff69ce630]
103:  4  /home/bouteill/parsec/master/parsec/build/parsec/libparsec.so.4(+0x7524c) [0x7ffff7ab324c]
103: Rank: 0, core 2: leaving PTG task (16, 10) of TP ptg_tp (2/0x934670), slept 1 seconds
103: Rank: 0, core 3: leaving PTG task (16, 9) of TP ptg_tp (2/0x934670), slept 1 seconds
103: Rank: 0, core 4: leaving PTG task (16, 8) of TP ptg_tp (2/0x934670), slept 1 seconds
103: Rank: 0, core 6: leaving PTG task (16, 7) of TP ptg_tp (2/0x934670), slept 1 seconds
```

**Comment by bosilca** [comment]
I have 4 tests consistently failing.
```
43 - collections/redistribute:mp (Timeout)
44 - collections/redistribute_random:mp (Timeout)
88 - dsl/dtd/pingpong:mp (Timeout)
99 - taskpool_wait (Failed)
```

They all see to fail in a similar fashion, so maybe the root is the same. I see the following trace 
```
[ 0] 0   libsystem_platform.dylib            0x000000018afbb4a4 _sigtramp + 56
[ 1] 0   libparsec.4.0.0.dylib               0x0000000102f9df90 parsec_list_lock + 28
[ 2] 0   libparsec.4.0.0.dylib               0x0000000102f9df90 parsec_list_lock + 28
[ 3] 0   libparsec.4.0.0.dylib               0x0000000102f9dee8 parsec_dtd_dequeue_taskpool + 64
[ 4] 0   libparsec.4.0.0.dylib               0x0000000102f8fc78 __parsec_taskpool_wait + 512
[ 5] 0   libparsec.4.0.0.dylib               0x0000000102f8fa58 parsec_taskpool_wait + 368
[ 6] 0   taskpool_wait                       0x0000000102b9ff40 main + 540
[ 7] 0   dyld                                0x0000000102d1508c start + 520
```

**Comment by bosilca** [comment]
Replaced by #411.

**Pull request closed without merging**